torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -437,20 +439,33 @@ def init(
     def get_handles_to_backward_prefetch(
         self,
         current_handles_key: _HandlesKey,
+        backward_prefetch: BackwardPrefetch,
     ) -> Optional[_HandlesKey]:
         """
         Returns the handles key of the handles to backward prefetch given the
         current handles key or ``None`` if there is no valid handles key to
         prefetch.
         """
-        current_index = self.handles_to_post_forward_order_index.get(current_handles_key, None)
-        if current_index is None:
-            return None
-        target_index = current_index - 1
-        if target_index < 0:
-            return None
-        target_handles_key = self.handles_post_forward_order[target_index]
-        return target_handles_key
+        if backward_prefetch == BackwardPrefetch.BACKWARD_PRE:
```

Comment:
This might be simpler to show the difference in the implementations:

```
handles = handles_to_post if BACKWARD_PRE else handles_to_pre

current_index = handles.get(...)
# rest of code
```